storage/innobase/include/fil0fil.h

```diff
@@ -52,6 +52,19 @@ Created 10/25/1995 Heikki Tuuri
 /** Structure containing encryption specification */
 struct fil_space_crypt_t;
 
+struct redo_log_key {
```

Comment:
This doesn't seem right. Redo log key defined in fil0fil.h.

Please put it in log0log.h

Reply:
I've put the rest of the code to fil0crypt.h, where the rest of keyring encryption is. The issue here is that fil0crypt includes fil0fil, and fil0fil defines fil_space_t, which needs a redo_log_key* data member. I could move everything to log0log, but I'm not sure it would be better than fil0crypt? Or maybe I could only forward declare it in fil0fil, I only need a pointer to it?

Reply:
just use forward declaration and put the redo_log_key code in log0log.h/.cc